runner/activity.go

```diff
@@ -41,7 +46,9 @@ func processActivityForServer(ctx context.Context, server *state.Server, globalC
 		}
 	}
 
-	if !globalCollectionOpts.ForceEmptyGrant {
+	if server.WebSocket.Load() != nil {
+		newGrant = *server.Grant.Load()
```

Comment:
There could be a case here where the websocket is initialized, but we didn't get the grant server message yet (or it wasn't processed yet by the Go routine).

Should we be blocking here, or falling back to HTTP if the websocket grant is not yet valid?